File: app/src/main/java/ru/application/homemedkit/databaseController/Medicine.java
package ru.application.homemedkit.databaseController;

import static ru.application.homemedkit.helpers.ConstantsHelper.BLANK;

import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "medicines")
public class Medicine {
    @PrimaryKey(autoGenerate = true)
    public long id = 0;
    public String cis = BLANK;
    public String productName = BLANK;
    public long expDate = -1L;
    public String prodFormNormName = BLANK;
    public String prodDNormName = BLANK;
    public double prodAmount = -1.0;
    public String phKinetics = BLANK;
    public String comment = BLANK;
    @Embedded
    public Technical technical = new Technical();

    public Medicine() {
    }

    @Ignore
    public Medicine(long id) {
        this.id = id;
    }

    @Ignore
    public Medicine(String cis,
                    String productName,
                    long expDate,
                    String prodFormNormName,
                    String prodDNormName,
                    double prodAmount,
                    String phKinetics,
                    String comment,
                    Technical technical) {
        this.cis = cis;
        this.productName = productName;
        this.expDate = expDate;
        this.prodFormNormName = prodFormNormName;
        this.prodDNormName = prodDNormName;
        this.prodAmount = prodAmount;
        this.phKinetics = phKinetics;
        this.comment = comment;
        this.technical = technical;
    }

    public Medicine(String cis,
                    String productName,
                    long expDate,
                    String prodFormNormName,
                    String prodDNormName,
                    String phKinetics,
                    Technical technical) {
        this.cis = cis;
        this.productName = productName;
        this.expDate = expDate;
        this.prodFormNormName = prodFormNormName;
        this.prodDNormName = prodDNormName;
        this.phKinetics = phKinetics;
        this.technical = technical;
    }

    @Ignore
    public Medicine(long id,
                    String cis,
                    String productName,
                    long expDate,
                    String prodFormNormName,
                    String prodDNormName,
                    double prodAmount,
                    String phKinetics,
                    String comment,
                    Technical technical) {
        this.id = id;
        this.cis = cis;
        this.productName = productName;
        this.expDate = expDate;
        this.prodFormNormName = prodFormNormName;
        this.prodDNormName = prodDNormName;
        this.prodAmount = prodAmount;
        this.phKinetics = phKinetics;
        this.comment = comment;
        this.technical = technical;
    }
}

File: app/src/main/java/ru/application/homemedkit/databaseController/Medicine.kt
package ru.application.homemedkit.databaseController

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE
import androidx.room.ForeignKey.Companion.SET_NULL
import androidx.room.PrimaryKey
import ru.application.homemedkit.helpers.BLANK

@Entity(
    tableName = "medicines",
    foreignKeys = [ForeignKey(
        entity = Kit::class,
        parentColumns = arrayOf("kitId"),
        childColumns = arrayOf("kitId"),
        onUpdate = CASCADE,
        onDelete = SET_NULL
    )]
)
data class Medicine(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0L,
    val kitId: Long? = null,
    val cis: String = BLANK,
    val productName: String = BLANK,
    val expDate: Long = -1L,
    val prodFormNormName: String = BLANK,
    val prodDNormName: String = BLANK,
    val prodAmount: Double = -1.0,
    val doseType: String = BLANK,
    val phKinetics: String = BLANK,
    val comment: String = BLANK,
    val image: String = BLANK,
    @Embedded
    val technical: Technical = Technical()
)